src/classes/TEST_RecurringDonationBuilder.cls

```diff
@@ -0,0 +1,363 @@
+        return this;
+    }    
+
+    /***
+    * @description Sets Day of Month
+    * @param dayOfMonth Day of Month value
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withDayOfMonth(String dayOfMonth) {
+        valuesByFieldName.put('Day_Of_Month__c', dayOfMonth);
+        return this;
+    }    
+
+    /***
+    * @description Sets number of Installments
+    * @param numberOfInstallments Number of Installments
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withInstallments(Integer numberOfInstallments) {
+        valuesByFieldName.put('npe03__Installments__c', numberOfInstallments);
+        return this;
+    }
+
+    /***
+    * @description Sets Installment Period to Yearly
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withInstallmentPeriodYearly() {
+        return withInstallmentPeriod(System.Label.npe03.RecurringDonationInstallmentPeriodYearly);
+    }
+
+    /***
+    * @description Sets Installment Period to Quarterly
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withInstallmentPeriodQuarterly() {
+        return withInstallmentPeriod(System.Label.npe03.RecurringDonationInstallmentPeriodQuarterly);
+    }
+
+    /***
+    * @description Sets Installment Period to Monthly
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withInstallmentPeriodMonthly() {
+        return withInstallmentPeriod(System.Label.npe03.RecurringDonationInstallmentPeriodMonthly);
+    }
+
+    /***
+    * @description Sets Installment Period to Weekly
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withInstallmentPeriodWeekly() {
+        return withInstallmentPeriod(System.Label.npe03.RecurringDonationInstallmentPeriodWeekly);
+    }
+
+    /***
+    * @description Sets Installment Period
+    * @param installmentPeriod Installment Period
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withInstallmentPeriod(String installmentPeriod) {
+        valuesByFieldName.put('npe03__Installment_Period__c', installmentPeriod);
+        return this;
+    }
+
+    /***
+    * @description Sets Schedule Type
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withScheduleTypeMultiplyValue() {
+        valuesByFieldName.put('npe03__Schedule_Type__c', System.Label.npe03.RecurringDonationMultiplyValue);
+        return this;
+    }
+
+    /***
+    * @description Sets Open Ended Status to Open Status
+    * @return TEST_RecurringDonationBuilder Builder instance
+    */
+    public TEST_RecurringDonationBuilder withOpenEndedStatusOpen() {
```

Comment:
Since `npe03__Open_Ended_Status__c` is an RD1 only field, there will need to be other methods for `withRecurringDonationType` and `withStatus` that update the RD2 specific fields. Plus, as I noted before if these methods are called for an RD2 format RD an exception should be thrown.